airflow/jobs/scheduler_job.py

```diff
@@ -1503,7 +1503,7 @@ def _do_scheduling(self, session) -> int:
                 )
                 .filter(
                     TI.dag_id.in_(list({dag_run.dag_id for dag_run in dag_runs})),
-                    TI.state.notin_(list(State.finished)),
+                    TI.state.notin_(list(State.finished) + [State.REMOVED]),
```

Comment:
I didn't add it to `State.finished` deliberately

https://github.com/apache/airflow/blob/419111696d58f0ed76b5cd6ab99db9f68478f11f/airflow/utils/state.py#L103-L118

coz the task is removed state is never executed